programs/temp/src/utils/offset.rs

```diff
@@ -0,0 +1,28 @@
+use crate::error::ErrorCode;
+use crate::seeds::OFFSET_METADATA_SEED;
+use crate::state::OffsetMetadata;
+use anchor_lang::prelude::*;
+
+// TODO: Double check if crate::ID is a correct owner of all things, or if we should use a PDA
```

Comment:
I would need to look deeper, but my intuition is it should be a pda derived from the global state address. This matches eg the mint authority in the main sunrise application:

https://github.com/sunrise-stake/app/blob/develop/programs/sunrise-stake/src/instructions/register_state.rs#L62